packages/side-nav/src/SideNavGroup.tsx

```diff
@@ -1,95 +1,115 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
 import { Transition } from 'react-transition-group';
-import ChevronRightIcon from '@leafygreen-ui/icon/dist/ChevronRight';
-import { css, cx } from '@leafygreen-ui/emotion';
-import { uiColors } from '@leafygreen-ui/palette';
-import { createDataProp, OneOf } from '@leafygreen-ui/lib';
+import {
+  createDataProp,
+  IdAllocator,
+  OneOf,
+  isComponentType,
+} from '@leafygreen-ui/lib';
 import { useUsingKeyboardContext } from '@leafygreen-ui/leafygreen-provider';
+import { isComponentGlyph } from '@leafygreen-ui/icon';
+import ChevronRight from '@leafygreen-ui/icon/dist/ChevronRight';
+import { prefersReducedMotion } from '@leafygreen-ui/a11y';
+import { uiColors } from '@leafygreen-ui/palette';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { spacing } from '@leafygreen-ui/tokens';
+import CollapsedSideNavItem from './CollapsedSideNavItem';
 import {
   ulStyleOverrides,
   sideNavItemSidePadding,
   sideNavWidth,
 } from './styles';
 
+const sideNavGroupIdAllocator = IdAllocator.create('side-nav-group');
+
 const button = createDataProp('side-nav-group-button');
 
-const sideNavLabelStyle = css`
+const listItemStyle = css`
+  & + & {
+    margin-top: ${spacing[2]}px;
+  }
+`;
+
+const labelStyle = css`
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
   font-size: 12px;
   letter-spacing: 0.3px;
   font-weight: bold;
   text-transform: uppercase;
   color: ${uiColors.green.dark2};
-  margin-top: 12px;
+  min-height: ${spacing[5]}px;
+  margin-top: 0;
   margin-bottom: 0;
   padding: 4px ${sideNavItemSidePadding}px 4px ${sideNavItemSidePadding}px;
-  line-height: 1.3em;
-  position: relative;
+  line-height: 1em;
```

Comment:
A curiosity rather than a comment, but why use em here when we use px everywhere else?

Reply:
By using em, we reduce the maintenance required if we update the font-size. I like em units when you specifically need something to be a whole number multiple of the current font-size.

Reply:
We use this in a fair number of places (14px font => 20px line-height, 16px font => 24px line-height) maybe we should consider applying this elsewhere. Again, out of scope for this PR but if that's a pattern we want to start applying, it's worth calling out

Reply:
The point of this is to eliminate line height. 1em line height is used to improve vertical centering usually where line wrapping isn't a major consideration. It's a CSS pattern not specific to our own standards, but general CSS authoring.